GVFS/GVFS.UnitTests.Windows/Windows/Upgrader/ProductUpgraderTests.cs

```diff
@@ -80,6 +80,74 @@ public void UpgradeAvailableOnSlowWhileOnLocalFastRing()
                 expectedUpgradeVersion:UpgradeTests.NewerThanLocalVersion);
         }
 
+        [TestCase]
+        public void RingInNugetFeedURLOverridesUpgradeRing()
+        {
+            // Pretend there is an upgrade available in Fast ring. Set upgrade.ring
+            // to fast and verify that Upgrader returns the new version.
+            Version newVersion;
+            string error;
+            this.SetUpgradeRing(GitHubUpgrader.GitHubUpgraderConfig.RingType.Fast.ToString());
+
+            // Replace pretend upgrade Release set by UpgradeTests.Setup() method
+            this.Upgrader.PretendNewReleaseAvailableAtRemote(UpgradeTests.NewerThanLocalVersion, GitHubUpgrader.GitHubUpgraderConfig.RingType.Fast);
+
+            this.Upgrader.TryQueryNewestVersion(out newVersion, out error).ShouldBeTrue();
+            newVersion.ShouldNotBeNull();

```

Comment:
Minor: You could make this validation a bit more specific and verify that `newVersion` is the same as `UpgradeTests.NewerThanLocalVersion`.

Reply:
(No need to block on this)